File: src/database/DBRoad.java
package database;

import model.Road;

import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.ArrayList;

/**
 * DBRoad
 * Handles all database functionality for roads.
 *
 * @author Dani Sander
 * @version 1.0
 * @since 20-05-15
 */

public class DBRoad {
    
    private DBConnection dbConnection;
    private static DBRoad instance;
    
    /**
     * Private constructor for singleton.
     * @throws SQLException 
     * @throws ClassNotFoundException 
     */
    private DBRoad() throws ClassNotFoundException, SQLException {
        dbConnection = DBConnection.getInstance();        
    }
    
    /**
     * Singleton method for class.
     * @return instance of class.
     * @throws SQLException 
     * @throws ClassNotFoundException 
     */
    public static DBRoad getInstance() throws ClassNotFoundException, SQLException {
        if (instance == null) {
            instance = new DBRoad();            
        }

        return instance;
    }

    public ArrayList<Road> getRoads() {
        ArrayList<Road> list;
        String sql = "select * from Road";
        list = (ArrayList<Road>) dbConnection.sendSQL(this, sql, "_formatRoad");
        return list;
    }

    public ArrayList<Road> _formatDefaultRoute(ResultSet rs) {
        ArrayList<Road> tableList = new ArrayList<>();
        try {
            while (rs.next()) {
                tableList.add(new Road(
                        rs.getLong("from"),
                        rs.getLong("to"),
                        rs.getDouble("distance"),
                        rs.getTime("time")
                ));
            }
        } catch (SQLException e) {
            // TODO Auto-generated catch block
            e.printStackTrace();
        }
        return tableList;
    }
}

File: Server/Database/DBRoad.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Model;
using System.Data;

namespace Server.Database
{
    public class DBRoad
    {
        public DBConnection DbConnection { get; private set; }
        private static DBRoad instance;

        /// private constructor for singelton     
        private DBRoad()
        {
            DbConnection = DBConnection.Instance;
        }

        /// singelton get instance method 
        /// returns the instance from DB 
        public static DBRoad Instance
        {
            get
            {
                if (instance == null)
                {
                    instance = new DBRoad();
                }

                return instance;
            }

        }

        /// <summary>
        /// Returns a list of all Roads.
        /// </summary>
        /// <param name="defaultDeliveryStopID"></param>
        /// <returns></returns>
        public List<Road> getRoads()
        {
            List<Road> list;
            String sql = "select * from Road";
            list = DbConnection.SendSQL<Road>(sql, ConvertToRoad);
            return list;
        }


        /// <summary>
        /// Takes the ReulstSet from database and returns a list of all Roads.
        /// </summary>
        /// <param name="rs"></param>
        /// <returns></returns>
        public List<Road> ConvertToRoad(IDataReader dataSet) 
        {
            List<Road> tableList = new List<Road>();
            while (dataSet.Read()) {
                tableList.Add(new Road(
                    dataSet.GetInt64(0),
                    dataSet.GetInt64(1),
                    dataSet.GetDouble(2),
                    (TimeSpan)dataSet.GetValue(3)
                ));
            }
            
            return tableList;
        }
    }
}